File: src/main/angular/src/app/models/file-properties-view.ts
import {DateTime} from 'luxon';
import {TransformDateTime} from '../misc/transform-date-time';

export class FilePropertiesView {
  uuid!: string;
  name!: string;
  mimeType!: string;
  sizeBytes!: number;
  parentUuid!: string | null;
  parentName!: string | null;

  @TransformDateTime()
  createdAt!: DateTime;

  @TransformDateTime()
  lastModifiedAt!: DateTime;
}

File: src/main/java/com/oppshan/files/file/FilePropertiesView.java
package com.oppshan.files.file;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.io.Serial;
import java.time.Instant;
import java.util.UUID;

@Valid
public record FilePropertiesView(
        @NotNull
        UUID uuid,

        @NotEmpty
        String name,

        @NotEmpty
        String mimeType,

        @PositiveOrZero
        long sizeBytes,

        UUID parentUuid,

        String parentName,

        @NotNull
        Instant createdAt,

        @NotNull
        Instant lastModifiedAt
) implements FileNodePropertiesView {

    @Serial
    private static final long serialVersionUID = 1L;
}